src/ExportMap.js

```diff
@@ -785,3 +763,32 @@ function makeSourceCode(text, ast) {
     return new SourceCode({ text, ast });
   }
 }
+
+/**
+ * Returns a project's TypeScript configuration.
+ */
+export function readTsConfig(context) {
```

Comment:
relying on hoisting is a bad practice; can we move this back up to where it was defined previously? that will have the side bonus of making the diff cleaner

Reply:
I'm also not a fan of relying on hoisting, but noted it was already in use in this file - and assumed it was a project convention, sorry. Moved back and will push an update soon.

Reply:
I haven't yet enabled the rule that forbids it; there's a lot of legacy coding patterns in the project to clean up. Thanks :-)